src/sentry/search/django/backend.py

```diff
@@ -250,6 +257,80 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        # TODO: This could also do something janky like switching based on the current
+        #       tagstore backend, or a global option, or a number of other things. Going
+        #       by the project seemed the best for now.
```

Comment:
Keep in mind that this limits our ability to use the `ServiceDelegator` to compare results side by side.

Reply:
Ya for sure. The problems I have are:

1. I will defend the awful tagstore API and why it exists any day
2. I do agree that "tags as the only entrypoint for Snuba" is probably bad. As in, Snuba does more than tags and search is a decent example.

So the only way to pull this behind the same service interface is to yank the *entire* search method into tagstore which feels kinda wrong. :( Or... another... service? Open to ideas? Would love ideas.

Reply:
Sorry, I should have been more specific in my comment. This was about the consideration for other methods of switching the backend in general, not specifically the tagstore backend or anything. Since search already is [implemented as a service](https://github.com/getsentry/sentry/blob/d2791de0c85c130bfa48549dc5538d8afe2f677b/src/sentry/search/__init__.py#L9-L13) we could use the delegation tooling, but only if we were able to run the entire `query` method in both contexts (one using the legacy implementation and one using the Snuba bits) and comparing results.

I think that the tagstore considerations are pretty much separate from my POV, with the exception that it would be kind of weird if we ran two searches, one that used the Snuba-backed tagstore backend and one that used the more broad Snuba search path.

Reply:
Arg, that's still weird because of `get_group_ids_for_search_filter`... if we're comparing/testing backends you could still have a scenario where Search: Django, Tagstore: Snuba, right? Which would call `get_group_ids_for_search_filter` and bomb out.

So I think you were right here that we should keep the `get_group_ids_for_search_filter` impl: https://github.com/getsentry/sentry/pull/7925#discussion_r181258792

Reply:
Yep, I agree with both comments.

I think we can probably figure out some way later to avoid getting into the search/tagstore comparison situation that you mentioned but I don't think there's really any significant harm in keeping `get_group_ids_for_search_filter` around either.